pkg/apis/pipeline/v1/container_validation.go

```diff
@@ -97,3 +103,291 @@ func RefNameLikeUrl(name string) error {
 	}
 	return nil
 }
+
+func validateStep(ctx context.Context, s Step, names sets.String) (errs *apis.FieldError) {
```

Comment:
+1 for colocating step and sidecar related functions in the container validation instead of task validation.